RecoLocalCalo/HcalRecProducers/plugins/alpaka/Mahi.dev.cc

```diff
@@ -701,6 +696,16 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE {
                   outputGPU.energyM0()[gch] = method0_energy;
                   outputGPU.timeM0()[gch] = time;
 
+                  // check as in cpu version if mahi is not needed
+                  // FIXME: KNOWN ISSUE: observed a problem when rawCharge and pedestal
+                  // are basically equal and generate -0.00000...
+                  // needs to be treated properly
+                  if (shrEnergyM0TotalAccum[lch] <= 0) {
+                    // do not need to run mahi minimization
+                    //outputEnergy[gch] = 0; energy already inited to 0
```

Comment:
Basically, this PR split the check in question in two separate checks, such that `shrEnergyM0TotalAccum[lch]` is checked only after the threads are synchronised.

I copy-pasted the comments in the code ("FIXME") a bit blindly. If others have feedback to improve this, I'm happy to do so.